cfpb/cf-tables
# cf-tables is ready for a PR!

This adds the bulk of cf-tables, built according to the new specifications for the CFPB design manual.

[Check out the auto-generated documentation!](http://mistergone.github.io/cf-tables/docs/)
## Additions
- `.table__small_stacked`, the base responsive table for small screens
- `.table__small_entry-header`, a small screen variation
- `.table-wrapper__scrolling`, a wrapper for another small screen variation
- `.table__striped`, a class that adds striped rows
- `.table__sortable`, a class that enables table sorting
- `cf-tables.js`, the JavaScript that enables table sorting
- Tests for `cf-tables.js`
## Testing
- `git clone` the repo
- Run `npm install`
- Run `bower install`
- Run `grunt`
- Check out `docs/index.html` for documentation
- Run `grunt test` to run qunit tests
## Review
- @Scotchester @ascott1 @jimmynotjim @niqjohnson and anyone else who wants to take a look
- **Feedback is appreciated!** I do not claim to be the master of all this stuff, so even tiny nitpicks are helpful!
## Todos
- ~~`cf-tables.js` should handle tables with `sorted_up` or `sorted_down` classes in their TRs by sorting them on document load.~~ _[Added here](https://github.com/mistergone/cf-tables/commit/efcdcdb9221f33621f0ae4e43bc69081a2377f7c)_
- `cf-tables.js` should be able to sort dates (I plan to add a `data-sort_type="date"` handler)
## Checklist
## Special Thanks

@ascott1 and @Scotchester have been immensely helpful in teaching me Capital Framework development, so hooray to them both!
## Animated GIF

Generating tables is like...
![xbqozc4](https://cloud.githubusercontent.com/assets/1490703/10168199/cabcca8a-6699-11e5-825b-b5899796c9d8.gif)
